src/device/device_ledger.cpp

```diff
@@ -1346,6 +1347,75 @@ namespace hw {
         return true;
     }
 
+    void device_ledger::get_transaction_prefix_hash(const cryptonote::transaction_prefix& tx, crypto::hash& h) {
+      AUTO_LOCK_CMD();
+      
+      int pref_length = 0, pref_offset = 0, offset = 0;
+
+      #ifdef DEBUG_HWDEVICE
+      crypto::hash h_x;
+      crypto::hash h_xx;
+      this->controle_device->get_transaction_prefix_hash(tx,h_x);
+      MDEBUG("get_transaction_prefix_hash [[IN]] h_x/1 "<<h_x);
+      #endif
+    
+      std::ostringstream s_x;
+      binary_archive<true> a_x(s_x);
+      ::serialization::serialize(a_x, const_cast<cryptonote::transaction_prefix&>(tx));
+      pref_length = s_x.str().size();
+      char pref[pref_length];
```

Comment:
Potential way for someone to pass a huge tx and smash the stack ?

Reply:
I dont know. Do you prefere a malloc/free based code?

Reply:
If the range s_x.str() is not bounded to a known size that's suitable for a stack, yes.

Reply:
ok, I will do